### DISCUSSION TITLE
Can you use glewlwyd as a proxy with a custom external authentication mechanism?

### DISCUSSION THREAD

**Discussion by ronchi** [babelouest/glewlwyd#212] [Q&A]
I have a situation where I'm using a legacy (and I mean truly old) SOAP process for authentication, but need to use that legacy authentication mechanism for newer software programs that use OAUTH2.  Obviously, the newer software applications won't use that kind of legacy authentication mechanism, so I was wondering if glewlwyd could be used as an authentication proxy, wherein the new software applications authenticate off glewlwyd using OAUTH2, but glewlwyd first queries the legacy authentication mechanism to determine if the use's credentials are valid (and then act accordingly).  Is that possible?

**Comment by babelouest** [comment]
Hello,

I don't know how SOAP authentication works in details.

A quick search gives me [this documentation](https://www.soapui.org/docs/soap-and-wsdl/authenticating-soap-requests/), so I guess one could implement a Glewlwyd authentication scheme where Glewlwyd would act as a proxy. But it's only a hypothesis.

At least, the authentication scheme architecture is designed to help these kind of goals.

You can check the [scheme documentation](https://github.com/babelouest/glewlwyd/tree/master/src/scheme) and try to implement your own authentication scheme.
